### DISCUSSION TITLE
Global Custom ExceptionMapper its not called when running in native code (Linux or Windows)

### DISCUSSION THREAD

**Discussion by hablocher** [quarkusio/quarkus#30372] [Q&A] [answered]
Hi folks! First time here!

I have the following problem:

My custom ExceptionMapper class is working fine when run with quarkus dev, but when i run native, Linux or Windows, the custom ExceptionMapper dont be called!

My code 

```
@Provider
public class CustomExceptionMapper implements ExceptionMapper<CustomException> {

	@Override
	public Response toResponse(CustomException exception) {
		return Response.status(Response.Status.BAD_REQUEST).entity(new ErrorMessage(exception.getMessages()))
				.build();
	}

}

```
Works like a charm running em development mode, but don't work in native mode.

Thanks for any help!

**Comment by stephan-strate** [comment, marked as answer]
Did you add `@RegisterReflection` to the ErrorMessage class?

**Reply by hablocher** [reply]
Stephan, thanks a lot! This is the solution!

But just to clarify, the annotation is `@RegisterForReflection` and i needed to mark another class in the code... this is the full code:


```
@Provider
@Named
public class CustomExceptionMapper implements ExceptionMapper<CustomException> {

	@Override
	public Response toResponse(CustomException exception) {
		if (exception instanceof HttpException) {
			ExceptionInfo exceptionInfo = exception.getStackOfMessages().get(0);
			return Response.status(Response.Status.fromStatusCode(Integer.parseInt(exceptionInfo.getErrorCode())))
					.entity(new ErrorMessage(exception.getMessages())).build();

		}
		return Response.status(Response.Status.BAD_REQUEST).entity(new ErrorMessage(exception.getMessages()))
				.build();
	}

}

```
Both classes, `ExceptionInfo ` and `ErrorMessage` needed to be annotaded

Thanks for your help! Peace!
